demo/src/test/java/com/example/demo/controller/DeveloperControllerTest.java

```diff
@@ -116,125 +120,106 @@ public void testUpdateDeveloper() {
-                "/users/"+ createdDeveloper.getId(),
-                HttpMethod.GET,
-                null,
-                Void.class
+    @Test
+    public void testDeveloperAndSquadInteraction() {
+        //Create Developer
+        Developer developer = developerRepo.save(generateRandomDeveloper());
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<Developer> createDeveloperRequest = new HttpEntity<>(developer, headers);
+        ResponseEntity<Developer> createDeveloperResponse = restTemplate.exchange(
+                "/developers",
+                HttpMethod.POST,
+                createDeveloperRequest,
+                Developer.class
         );
 
+        Developer createdDeveloper = createDeveloperResponse.getBody();
+        assertThat(createdDeveloper).isNotNull();
+        assertThat(createdDeveloper.getDeveloperId()).isNotNull();
 
-        assertThat(getResponseEntity.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
+        //create squad
+        Squad squad = squadRepo.save(generateRandomSquad());
+        HttpEntity<Squad> createSquadRequest = new HttpEntity<>(squad, headers);
+        ResponseEntity<Squad> createSquadResponse = restTemplate.exchange(
+                "/squad",
+                HttpMethod.POST,
+                createSquadRequest,
+                Squad.class
+        );
 
+        Squad createdSquad= createSquadResponse.getBody();
+        assertThat(createdSquad).isNotNull();
+        assertThat(createdSquad.getSquadId()).isNotNull();
 
 
+        //add developer from the squad
+        HttpEntity<Void> addDeveloperToSquadRequest = new HttpEntity<>(headers);
+        ResponseEntity<Squad> addDeveloperToSquadResponse = restTemplate.exchange(
+                "/squad/"+ createdSquad.getSquadId()+"/add-developer/"+createdDeveloper.getDeveloperId(),
+                HttpMethod.PUT,
+                addDeveloperToSquadRequest,
+                Squad.class
+        );
 
+        Squad squadWithDeveloper = addDeveloperToSquadResponse.getBody();
+        assertThat(squadWithDeveloper).isNotNull();
+        MatcherAssert.assertThat(squadWithDeveloper.getDevelopers(), Matchers.hasItem(createdDeveloper));
 
+        // Remove the developer from the squad
+        HttpEntity<Void> removeDeveloperFromSquadRequest = new HttpEntity<>(headers);
+        ResponseEntity<Squad> removeDeveloperFromSquadResponse = restTemplate.exchange(
+                "/squad/" + createdSquad.getSquadId() + "/remove-developer/" + createdDeveloper.getDeveloperId(),
+                HttpMethod.PUT,
+                removeDeveloperFromSquadRequest,
+                Squad.class
+        );
+        Squad squadWithoutDeveloper = removeDeveloperFromSquadResponse.getBody();
+        assertThat(squadWithoutDeveloper).isNotNull();
+        assertThat(squadWithoutDeveloper.getDevelopers()).isNotIn(createdDeveloper);
+    }
```

Comment:
All your tests are failing

I'd suggest to do following:
1. Start with unit tests. Implement all missing unit tests first to cover 100% of code
2. Then try and investigate why those tests are failing
3. Create SquadControllerTests